Source/RapyutaSimulationPlugins/Private/Core/RRCoreUtils.cpp

```diff
@@ -19,20 +20,36 @@
 #include "Core/RRTypeUtils.h"
 #include "Core/RRUObjectUtils.h"
 
-const TMap<ERRFileType, const TCHAR*> URRCoreUtils::SimFileExts = {
-    {ERRFileType::NONE, EMPTY_STR},
-    {ERRFileType::UASSET, TEXT(".uasset")},
-    {ERRFileType::INI, TEXT(".ini")},
-    {ERRFileType::YAML, TEXT(".yaml")},
-    {ERRFileType::IMAGE_JPG, TEXT(".jpg")},
-    {ERRFileType::IMAGE_PNG, TEXT(".png")},
-    {ERRFileType::IMAGE_EXR, TEXT(".exr")},
-    {ERRFileType::IMAGE_HDR, TEXT(".hdr")},
-    {ERRFileType::URDF, TEXT(".urdf")},
-    {ERRFileType::SDF, TEXT(".sdf")},
-    {ERRFileType::GAZEBO_WORLD, TEXT(".world")},
-    {ERRFileType::MJCF, TEXT(".mjcf")},
-};
+IImageWrapperModule* URRCoreUtils::SImageWrapperModule = nullptr;
+TMap<ERRFileType, TSharedPtr<IImageWrapper>> URRCoreUtils::SImageWrappers;
+
+const TMap<ERRFileType, const TCHAR*> URRCoreUtils::SimFileExts = {{ERRFileType::NONE, EMPTY_STR},
```

Comment:
Right, I just want to avoid a bit unwieldy static_cast<uint8> from the enum value. Given the map size as the no of file types being not so large, its overheads should be acceptable I suppose.